src/main/java/com/wolox/training/models/User.java

```diff
@@ -43,6 +43,14 @@ public User() {
     books = new ArrayList<>();
   }
 
+  public User(String userName, String name, LocalDate birthDate,
+      List<Book> books) {
+    this.userName = userName;
+    this.name = name;
+    this.birthDate = birthDate;
+    this.books = books;
```

Comment:
Like before, you should use your setters here instead of this assignation. This way you assure that the preconditions are accomplished :)